tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java

```diff
@@ -270,7 +248,7 @@ public TaskDTO unassignTask(final String taskId) {
     final TaskEntity taskEntity = taskStore.persistTaskUnclaim(taskBefore);
     if (taskBefore.getImplementation().equals(TaskImplementation.ZEEBE_USER_TASK)) {
       try {
-        zeebeClient.newUserTaskUnassignCommand(taskBefore.getKey()).send().join();
+        taskMutationAdapter.unassignUserTask(taskEntity);
       } catch (final ClientException exception) {
         taskStore.persistTaskClaim(taskBefore, taskBefore.getAssignee());
```

Comment:
we may need to catch the runtime exceptions thrown by the broker to apply this logic 🤔